public/app/core/components/NavBar/Next/NavBarItem.tsx

```diff
@@ -107,7 +107,7 @@ const NavBarItem = ({
                 <Item key={getNavModelItemKey(item)} textValue={item.text}>
                   <NavBarMenuItem
                     isDivider={!isSection && item.divider}
-                    icon={isSection ? undefined : (item.icon as IconName)}
+                    icon={item.showIconInNavbar && !isSection ? (item.icon as IconName) : undefined}
```

Comment:
nit: I think it'd be easier to read / scan if this was defined as a `const` as you're doing for `translationKey`, `itemText` etc.